### PR TITLE
feat: cdk deployment, nodejs 10->12 upgrade

### PR DESCRIPTION
**Pull request by philvarner**

### CODE REVIEW

File: DEPLOYMENT.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+Connect to the EC2 instance.
+
+These command assume AWS credentials have been configured in the environment, either via the 
+`AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variables or via `~/.aws/credentials`.
+
+Configure these shell variables:
+
+```bash
+export GCC_PERMISSIONS_BOUNDARY_ENABLED=False
```
Comment (philvarner) [inline DEPLOYMENT.md]: I copied this from one of the other apis, but shouldn't this be setting it to True for GCC deploy?

File: cdk/stack.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        self,
+        scope: core.Construct,
+        id: str,
+        stack_name: str,
+        memory: int = 512,
+        timeout: int = 30,
+        entry: str = ".",
+        handler: str = "lib/lambda.handler",
+        env: Dict = None,
+        runtime: lambda_.Runtime = lambda_.Runtime.NODEJS_12_X,
```
Comment (philvarner) [inline cdk/stack.py:48]: nodejs10.x is stage 1 EOL at the end of July and stage 2 EOL at end of Aug, so upgrading to 12.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM lambci/lambda:build-nodejs12.x
+
+WORKDIR /tmp
+
+COPY . .
+
+RUN npm pack --unsafe-perm
```
Comment (philvarner) [inline Dockerfile:8]: when running in a Docker container as root, this flag is necessary to force `prepare` to run, which in our config runs `build`.

File: cdk/stack.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            id=f"{stack_name}-backend",
+            code=self.create_package(entry),
+            handler=handler,
+            runtime=runtime,
+            environment=env,
+            memory_size=memory,
+            timeout=core.Duration.seconds(timeout),
+        )
+
+        api = apigw.LambdaRestApi(
```
Comment (philvarner) [inline cdk/stack.py:67]: this uses the "legacy" rest api, same as the terraform definitions. If using an apigatewayV2 HttpApi (as some of our other APIs use), the event format is different, so the lambda handler fails.

File: DEPLOYMENT.md (hunk 1 above)
Comment (abarciauskas-bgse) [reply DEPLOYMENT.md]: Yes

Comment (abarciauskas-bgse) [reply DEPLOYMENT.md]: Yes - perhaps just add a comment "set to True if deploying to GCC"

**Review by abarciauskas-bgse — APPROVED** [review]
Looks reasonable, also verified deployment at https://ngygtotzg9.execute-api.us-west-2.amazonaws.com/prod/file-staging/nasa-map/ATL08_ARD-beta___001/global/ept/ept-tileset/tileset.json

**Pull request merged**